python/cudf/cudf/_lib/pylibcudf/scalar.pyx

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+cimport pyarrow.lib
+from cython cimport no_gc_clear
+from libcpp.memory cimport unique_ptr
+
+import pyarrow.lib
+
+from rmm._lib.memory_resource cimport get_current_device_resource
+
+from cudf._lib.cpp.scalar.scalar cimport fixed_point_scalar, scalar
+from cudf._lib.cpp.wrappers.decimals cimport (
+    decimal32,
+    decimal64,
+    decimal128,
+    scale_type,
+)
+
+from .types cimport DataType, type_id
+
+
+# The DeviceMemoryResource attribute could be released prematurely
+# by the gc if the DeviceScalar is in a reference cycle. Removing
+# the tp_clear function with the no_gc_clear decoration prevents that.
+# See https://github.com/rapidsai/rmm/pull/931 for details.
+@no_gc_clear
+cdef class Scalar:
+    """A scalar value in device memory."""
+    # Unlike for columns, libcudf does not support scalar views. All APIs that
+    # accept scalar values accept references to the owning object rather than a
+    # special view type. As a result, pylibcudf.Scalar has a simpler structure
+    # than pylibcudf.Column because it can be a true wrapper around a libcudf
+    # column
+
+    def __cinit__(self, *args, **kwargs):
+        self.mr = get_current_device_resource()
+
+    def __init__(self, pyarrow.lib.Scalar value=None):
+        # TODO: This case is not something we really want to
+        # support, but it here for now to ease the transition of
+        # DeviceScalar.
+        if value is not None:
+            raise ValueError("Scalar should be constructed with a factory")
+
+    @staticmethod
+    def from_pyarrow_scalar(pyarrow.lib.Scalar value, DataType data_type=None):
+        # Allow passing a dtype, but only for the purpose of decimals for now
+
+        # Need a local import here to avoid a circular dependency because
+        # from_arrow_scalar returns a Scalar.
+        from .interop import from_arrow_scalar
+
+        cdef Scalar s = from_arrow_scalar(value)
```

Comment:
We should eliminate one of `Scalar.from_pyarrow_scalar` or `interop.from_arrow_scalar`. Currently they are both part of the "public" pylibcudf API, and it's not clear when one should use which. It seems that `Scalar.from_pyarrow_scalar()` is the correct way to construct a `Scalar` object, while the `interop.from_arrow_scalar` is just a helper for the former.

Reply:
Subsuming `interop.from_arrow_scalar` here will have the added advantage of not needing a function-local import.

Reply:
This is done.

Reply:
For consistency I've done the same for Table.